### ISSUE TITLE
IDISP003. Private EIMI 

### ISSUE THREAD

**Issue by kuzyaka** [DotNetAnalyzers/IDisposableAnalyzers#83]
In this case Dispose could not be called explicitly. I guess it is false positive IDISP0003

class SomeClass : IDisposable
{
    public Close() { ... }
    private void IDisposable.Dispose()
    {
      this.Close();
    }
}
...
SomeClass some = new SomeClass();
some = null; // false positive IDISP003

**Comment by JohanLarsson** [comment]
`SomeClass` above does not compile, did you mean:
```cs
class SomeClass : IDisposable
{
    public void Close() {  }
    void IDisposable.Dispose()
    {
        this.Close();
    }
}
```

If so it is a tricky spot imo, not sure what is best there. As the Dispose implementation is explicit it is a sign that author meant something spexial there.

**Comment by kuzyaka** [comment]
> If so it is a tricky spot imo, not sure what is best there. As the Dispose implementation is explicit it is a sign that author meant something spexial there.

This could be binary reference and IDisposableAnanlyzers user could only ignore this warning.
In this case this warn useless

**Comment by JohanLarsson** [comment]
>This could be binary reference and IDisposableAnanlyzers user could only ignore this warning.
In this case this warn useless

I don't understand.

**Comment by JohanLarsson** [comment]
Have you seen [gitter](https://gitter.im/DotNetAnalyzers/IDisposableAnalyzers) btw?

**Comment by kuzyaka** [comment]
This case is found in RabbitMQ.Client.MessagePatterns.Subscription (nuget package RabbitMQ.Client)
Not sure I should explicitly call Dispose() via casting to IDisposable

**Comment by JohanLarsson** [comment]
Yes it is tricky, if you use it in a `using` with will call the explicit implementation.

**Comment by kuzyaka** [comment]
"Using" is not  always applicable. It may be used for "in-place" operations, that do not require saved state.

For example I have method "GetActiveSuscription" that check current subscription state and re-created it when it is closed or lost. I cannot use "using" here
